docsrc/markdown/authentication.md

```diff
@@ -38,6 +38,33 @@ async def run():
         logging.info(await auth_session.session_token)
 ```
 
+### Authentication using private key content
+Instead of configuring the path of RSA private key config file, you can also authenticate the bot 
+and extension app by using directly the private key or certificate content. This feature is useful when either 
+RSA private key is fetched from an external secrets storage. The code snippet below will give you 
+an example showing how to set directly the private key content to the Bdk configuration for authenticating the bot.
+````python
+import asyncio
+import logging
+
+from symphony.bdk.core.config.loader import BdkConfigLoader
+from symphony.bdk.core.symphony_bdk import SymphonyBdk
+
+async def run():
+    # loading configuration
+    config = BdkConfigLoader.load_from_symphony_dir("config.yaml")
+    async with SymphonyBdk(config) as bdk:
+        # update private key with content
+        private_key_string = '-----BEGIN RSA PRIVATE KEY-----\n'\
+                             'zI2OZtdb8fu/xl7itIAOzKLFg3mhA...\n'\
+        config.setBotConfig(private_key_content=private_key_string)
```

Comment:
is the setBotConfig pythonic?
Should it be set_bot_config?

do we want to provide such? should not it be rather an example with a config file?

Reply:
You're right, I put it to set_bot_config. 
Do you mean that I need to create an example for this and keep this one for config file usage ?